src/display_context.hpp

```diff
@@ -57,6 +57,8 @@ class display_context {
 	virtual const std::vector<std::string> & hidden_label_categories() const = 0;
 	std::vector<std::string> & hidden_label_categories_ref();
 	const team& get_team(int side) const;
+
+	template<typename T = void>
```

Comment:
It's probably not worth considering whether or not it gets inlined. I doubt any performance gain from inlining it vs not inlining it would be statistically significant.